### ISSUE TITLE
Improve how we upgrade packages. 

### ISSUE THREAD

**Issue by jameshadfield** [nextstrain/auspice#1462]
Keeping auspice's libraries up-to-date, primarily in relation to security concerns but also for general package updates, has become a major chore and is holding us back. Currently 17 of the 41 open PRs are from dependebot or our manual attempts to upgrade packages. 

The biggest blocker is that changes can cause nextstrain.org (and, presumably other applications which extend auspice) to fail to build ([example 1](https://github.com/nextstrain/auspice/pull/1376), [example2](https://github.com/nextstrain/auspice/pull/1436)). 

This issue doesn't cover changes, such as upgrading webpack versions, where I think manual work will always be necessary.

**Comment by tsibley** [comment]
Some thoughts to start discussion…

How are we being held back by the current process of using known-good/locked versions of deps? How would we be propelled forward by continually closely tracking the latest versions?

Is the issue PR overload (i.e. the 17/41 ratio)? We could turn off dependabot. Or change it to run on a more infrequent schedule (quarterly?) instead of as-it-happens.

Not all updates are made equal. Security updates are more important, of course, but even amongst those not all security updates apply to our usage. We've also dodged a few malicious package releases because we didn't immediately/automatically update to them before they were flagged as bad.

**Comment by jameshadfield** [comment]
Yeah, I'm not a fan of staying continually up-to-date for the sake of it. I'd be strongly in favour of a quarterly dependebot schedule. 

But we also need to work out why those examples (linked above) don't build on nextstrain.org, as this effectively stops us accepting package upgrade recommendations.

**Comment by tsibley** [comment]
Ah, those toggles you screenshotted are high-level switches, but there's [a lot of configuration](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates) that can be done using a `.github/dependabot.yml` file, including setting a schedule.

**Comment by tsibley** [comment]
(Although monthly is the longest interval. I'm also not sure if you can have different schedules for security vs. non-security updates.)

**Comment by victorlin** [comment]
> I'm also not sure if you can have different schedules for security vs. non-security updates.

I just looked into this a bit more, and I don't think there is even a way to set a schedule on security updates. Only non-security updates (termed "version updates" on the GitHub docs) can be scheduled. I can't find it documented, but from my experience the security updates are at least checked on every change pushed to the default branch.

The [configuration options overview table](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#configuration-options-for-the-dependabotyml-file) notes that `schedule.interval` only applies to version updates (e.g. regular checking for new Auspice versions on downstream repos):

![image](https://user-images.githubusercontent.com/13424970/194425997-b07bb2a2-78bc-4916-aec4-82415585c2f1.png)

So our options for handling Dependabot security updates are:

1. Do nothing about the PRs (status quo)
2. Manually review each PR (reduces PR clutter but may require tedious testing?)
3. Auto-merge some/all PRs (reduces PR clutter but comes with a chance of untested breakage)
4. Disable the feature [in repo settings](https://github.com/nextstrain/auspice/settings/security_analysis) (reduces PR clutter but no visibility to security issues)

**Comment by tsibley** [comment]
> 4\. Disable the feature [in repo settings](https://github.com/nextstrain/auspice/settings/security_analysis) (reduces PR clutter but no visibility to security issues)

Won't security issues still get flagged on the repo (and sent as notifications per individual user pref), just not as dependabot PRs?

**Comment by victorlin** [comment]
Oh you're right. We can keep **Dependabot alerts** enabled and disable **Dependabot security updates**. I've updated my previous comment.

![dependabot](https://user-images.githubusercontent.com/13424970/195656575-f02d1ab4-725e-483d-8953-54bc57eafad0.png)

**Comment by victorlin** [comment]
The latest 9 open PRs are dependabot security updates. We have not merged one since #1159 and I don't think there is any reason to change. I vote to (4) disable Dependabot security updates to reduce PR clutter.

**Comment by victorlin** [comment]
> (4) disable Dependabot security updates

This has been disabled in the nextstrain.org repo for a while ([last PR was in 2023](https://github.com/nextstrain/nextstrain.org/issues?q=label%3Adependencies%20"static-site")) and was just disabled for auspice.us for https://github.com/nextstrain/auspice.us/issues/144.

**Comment by joverlee521** [comment]
FWIW, Dependabot security updates have essentially been disabled in this repo because they are paused. The last [Dependabot Updates](https://github.com/nextstrain/auspice/actions/workflows/dependabot/dependabot-updates) action is from over a year ago on Mar 17, 2025. 

<img width="1379" height="310" alt="Image" src="https://github.com/user-attachments/assets/cbd5e0f1-ead5-43db-a99d-cb8684332587" />

---

I've disabled Dependabot security updates to make it official.
